core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -651,6 +654,49 @@ protected Optional<ConnectorExpression> visitCast(Cast node, Void context)
             return Optional.empty();
         }
 
+        @Override
+        protected Optional<ConnectorExpression> visitSimpleCaseExpression(SimpleCaseExpression node, Void context)
```

Comment:
Please also add a translation from `Call` back to `SimpleCaseExpression` at `ConnectorToSqlExpressionTranslator#translateCall`

Reply:
Addressed comments. PTAL @assaf2